src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -192,37 +188,18 @@ public Money EstimateFee(TransactionBuildContext context)
         /// Initializes the context transaction builder from information in <see cref="TransactionBuildContext"/>.
         /// </summary>
         /// <param name="context">Transaction build context.</param>
-        private void InitializeTransactionBuilder(TransactionBuildContext context)
+        public virtual void InitializeTransactionBuilder(TransactionBuildContext context)
         {
             Guard.NotNull(context, nameof(context));
             Guard.NotNull(context.Recipients, nameof(context.Recipients));
             Guard.NotNull(context.AccountReference, nameof(context.AccountReference));
 
-            context.TransactionBuilder = new TransactionBuilder(this.Network)
-            {
-                CoinSelector = new DefaultCoinSelector
-                {
-                    GroupByScriptPubKey = context.UseAllInputs
-                },
-                // Smart contract calls allow dust. Should be an option on TransactionBuildContext in future.
-                DustPrevention = context.DustPrevention,
-                StandardTransactionPolicy = this.transactionPolicy
-            };
-
             this.AddRecipients(context);
             this.AddOpReturnOutput(context);
             this.AddCoins(context);
             this.AddSecrets(context);
-            // Also added for smart contract demo
-            if (!context.ChangeAddressSet)
-            {
+            if (context.ChangeAddress == null)
```

Comment:
I think we didn't have this check before.

Reply:
If I recall correctly, you're right the FN didn't, but you also couldn't select which address to send your change to by setting a ChangeAddress in the TransactionBuildContext that gets passed into BuildTransaction. 

I think it's a little confusing that the TransactionBuildContext that I build and pass into BuildTransaction is then continually modified by the WalletTransactionHandler + Builder.

An alternative approach would be to have a kind of `TransactionBuildOptions` object that is passed into the TransactionBuilder. TransactionBuildContext would only be used internally. I'd really like to set this up and it would help both FN and SC a lot I think @dangershony what do you reckon? Can I work on this?

Reply:
@codingupastorm just so that we can move this along, I would suggest that we get this merged and you can work on this in a separate PR? BTW I agree with your suggestion 👍

Reply:
Also agree on a TransactionBuildOptions yeah explore this possibility